src/rulesets/asyncapi/index.json

```diff
@@ -0,0 +1,435 @@
+    "asyncapi2-payload": {
+      "description": "Payloads must be valid against AsyncAPI Schema object.",
+      "message": "{{error}}",
+      "severity": 0,
+      "recommended": true,
+      "type": "validation",
+      "given": [
+        "$.components.messageTraits[?(@.schemaFormat === void 0)].payload",
+        "$.components.messages[?(@.schemaFormat === void 0)].payload",
+        "$.channels.*.[publish,subscribe][?(@property === 'message' && @.schemaFormat === void 0)].payload"
+      ],
+      "then": {
+        "function": "asyncApi2PayloadValidation"
+      }
+    },
+    "asyncapi2-schema-default": {
+      "description": "Default must be valid against its defined schema.",
+      "message": "{{error}}",
+      "severity": 0,
+      "recommended": true,
+      "type": "validation",
+      "given": [
+        "$.components.schemas.*.default^",
+        "$.components.parameters.*.schema.default^",
+        "$.channels.*.parameters.*.schema.default^"
+      ],
+      "then": {
+        "function": "schemaPath",
+        "functionOptions": {
+          "field": "default",
+          "schemaPath": "$",
+          "allErrors": true
+        }
+      }
+    },
+    "asyncapi2-schema-example": {
+      "description": "Examples must be valid against their defined schema.",
+      "message": "{{error}}",
+      "severity": 0,
+      "recommended": true,
+      "type": "validation",
+      "given": [
+        "$.components.schemas.*.example^",
+        "$.components.parameters.*.schema.example^",
+        "$.channels.*.parameters.*.schema.example^"
+      ],
+      "then": {
+        "function": "schemaPath",
+        "functionOptions": {
+          "field": "example",
+          "schemaPath": "$",
+          "allErrors": true
+        }
+      }
+    },
+    "asyncapi2-schema-examples": {
+      "description": "Examples must be valid against their defined schema.",
+      "message": "{{error}}",
+      "severity": 0,
+      "recommended": true,
+      "type": "validation",
+      "given": [
+        "$.components.schemas.*.examples^",
+        "$.components.parameters.*.schema.examples^",
+        "$.channels.*.parameters.*.schema.examples^"
+      ],
+      "then": {
+        "function": "schemaPath",
+        "functionOptions": {
+          "field": "$.examples.*",
+          "schemaPath": "$",
+          "allErrors": true
+        }
+      }
+    },
+    "asyncapi2-schema": {
+      "description": "Validate structure of AsyncAPI v2.0.0 Specification.",
+      "message": "{{error}}",
+      "severity": 0,
+      "recommended": true,
+      "type": "validation",
+      "given": "$",
+      "then": {
+        "function": "schema",
+        "functionOptions": {
+          "schema": {
+            "$ref": "./schemas/schema.asyncapi2.json"
+          },
+          "allErrors": true
+        }
+      }
+    },
+    "asyncapi2-server-no-empty-variable": {
+      "description": "Server URL should not have empty variable substitution pattern.",
```

Comment:
the awesome rule would be that checks that you have a variable in your server url {port} but you do not describe it with variable object -> https://www.asyncapi.com/docs/specifications/2.0.0/#serverVariableObject which make the server url useless

Reply:
another awesome valid rule is for the same use case, for variables but in channel names

Reply:
@deberg This is on my wishlist as well. This kind of rule would require a custom function. I've updated the description of that PR to keep track of these rules and add them through another PR. If you can think of other rules that would make sense, please share them!